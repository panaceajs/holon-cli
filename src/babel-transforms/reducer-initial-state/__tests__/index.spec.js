const pluginTester = require('babel-plugin-tester');
const reducerInitialState = require('../');
const pluginSyntaxObjectRestSpread = require('@babel/plugin-syntax-object-rest-spread');
const dynamicImport = require('@babel/plugin-syntax-dynamic-import');

pluginTester({
  filename: __filename,
  plugin: reducerInitialState,
  snapshot: true,
  babelOptions: {
    sourceType: 'module',
    plugins: [dynamicImport, pluginSyntaxObjectRestSpread]
  },
  tests: [
    {
      title:
        'it should add action types to new state namespace in case of variable assignment',
      fixture: '__fixtures__/with-assignment.js',
      pluginOptions: {
        actionTypes: { foo: 'foo' },
        stateNamespace: 'nonExistingNamespace',
        stateKey: 'state'
      }
    },
    {
      title:
        'it should complete action types to existing state namespace in case of variable assignment',
      fixture: '__fixtures__/with-assignment.js',
      pluginOptions: {
        actionTypes: { foo: 'foo', existing: 'existing' },
        stateNamespace: 'existingNamespace',
        stateKey: 'state'
      }
    }
  ]
});
