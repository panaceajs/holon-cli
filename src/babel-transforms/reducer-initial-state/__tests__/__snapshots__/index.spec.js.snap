// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`holon-reducer-initial-state it should add action types to new state namespace in case of variable assignment: it should add action types to new state namespace in case of variable assignment 1`] = `
"
export default (
  state = {
    existing: 'existing'
  }
) => {};

      ↓ ↓ ↓ ↓ ↓ ↓

export default ((state = {
  existing: 'existing',
  foo: 'foo-initial-value'
}) => {});
"
`;

exports[`holon-reducer-initial-state it should complete action types to existing state namespace in case of variable assignment: it should complete action types to existing state namespace in case of variable assignment 1`] = `
"
export default (
  state = {
    existing: 'existing'
  }
) => {};

      ↓ ↓ ↓ ↓ ↓ ↓

export default ((state = {
  existing: 'existing',
  foo: 'foo-initial-value'
}) => {});
"
`;
